alf/utils/metric_utils.py

```diff
@@ -37,14 +37,17 @@ def __init__(self, time_step_spec, action_spec, policy_state_spec,
         super(Policy, self).__init__(
             time_step_spec=time_step_spec,
             action_spec=action_spec,
-            policy_state_spec=policy_state_spec)
+            policy_state_spec=(policy_state_spec, action_spec))
         # attribute named `_action_fn` already exist in parent
         self._action_fn1 = action_fn
 
     def _action(self, time_step, policy_state=(), seed=None):
-        policy_step = self._action_fn1(time_step, policy_state)
```

Comment:
`reset_state_if_necessary` for new episode .  And is there better way to store `prev_action` ?  it's very strange to store in `state`

Reply:
tfa_eager_compute() handles that.

Reply:
I agree. But this is the easiest way to add prev_action using tfa_eager_compute().
Perhaps we should use OnPolicyDriver for eval too.